network/setup-ip

```diff
@@ -188,36 +209,36 @@ fi
 
 if [ "$ACTION" == "add" ]; then
     MAC="$(get_mac_from_iface "$INTERFACE")"
-    if [ -n "$MAC" ]; then
-        ip="$(/usr/bin/qubesdb-read "/net-config/$MAC/ip" 2> /dev/null)" || ip=
-        ip6="$(/usr/bin/qubesdb-read "/net-config/$MAC/ip6" 2> /dev/null)" || ip6=
-        netmask="$(/usr/bin/qubesdb-read "/net-config/$MAC/netmask" 2> /dev/null)" || netmask=
-        netmask6="$(/usr/bin/qubesdb-read "/net-config/$MAC/netmask6" 2> /dev/null)" || netmask6=
-        gateway="$(/usr/bin/qubesdb-read "/net-config/$MAC/gateway" 2> /dev/null)" || gateway=
-        gateway6="$(/usr/bin/qubesdb-read "/net-config/$MAC/gateway6" 2> /dev/null)" || gateway6=
-
-        # Handle legacy values
-        LEGACY_MAC="$(/usr/bin/qubesdb-read /qubes-mac 2> /dev/null)" || LEGACY_MAC=
-        if [ "$MAC" == "$LEGACY_MAC" ] || [ -z "$LEGACY_MAC" ]; then
-            if [ -z "$ip" ]; then
-                ip="$(/usr/bin/qubesdb-read /qubes-ip 2> /dev/null)" || ip=
-            fi
-            if [ -z "$ip6" ]; then
-                ip6="$(/usr/bin/qubesdb-read /qubes-ip6 2> /dev/null)" || ip6=
-            fi
-            if [ -z "$gateway" ]; then
-                gateway="$(/usr/bin/qubesdb-read /qubes-gateway 2> /dev/null)" || gateway=
-            fi
-            if [ -z "$gateway6" ]; then
-                gateway6="$(/usr/bin/qubesdb-read /qubes-gateway6 2> /dev/null)" || gateway6=
-            fi
+    prefix="/net-config/$MAC"
+    if [[ -n "$MAC" ]]; then
+        # prefix begins with / so -- is not needed
+        if /usr/bin/qubesdb-read "${prefix}custom" > /dev/null 2>&1; then
+            custom=true
+        elif [[ "$?" = '2' ]]; then
+            custom=false
+        else
+            echo "Could not check if ${prefix}custom exists!" >&2
+            exit 1
         fi
-
-        if [ -z "$netmask" ]; then
-            netmask="255.255.255.255"
+        if ip4=$(exec /usr/bin/qubesdb-read "${prefix}ip" 2>/dev/null); then
+            :
+        elif [[ "$?" != '2' ]]; then
+            echo "Could not check if /net-config/$MAC/ip exists!" >&2
+            exit 1
+        else
+            prefix=/qubes-
+            ip4=$(exec /usr/bin/qubesdb-read "${prefix}ip")
         fi
-        if [ -z "$netmask6" ]; then
-            netmask6="128"
+        netmask=$(exec /usr/bin/qubesdb-read -e 255.255.255.255 "${prefix}netmask")
```

Comment:
`qubesdb-read` doesn't have `-e` option (few instances below too), is it some your local change? Looks like "default value", which sounds like a useful feature, but I'm not sure if `-e` is the most intuitive option for that, I'd rather choose `-d`.